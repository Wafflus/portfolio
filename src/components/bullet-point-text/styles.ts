import styled from "styled-components";

export const BulletPointTextStyled = styled.li`
    color: ${({ theme }) => theme.data.text.lightColor};
    font-family: '${({ theme }) => theme.data.fontFamilies.outfit}', sans-serif;
    font-size: 1.125rem;
    letter-spacing: 0.1rem;
    line-height: 120%;
    list-style-type: none;
    position: relative;
    text-align: center;
    transition: transform 0.5s;

    &:hover {
        transform: scale(102%);
    }
    
    ${props => props.theme.breakpoint.above('lg',
        `
            &::before {
                background-color: ${props.theme.data.secondaryColor};
                border-radius: 50%;
                content: '';
                height: 20px;
                left: -40px;
                position: absolute;
                transition: background-color 0.5s;
                width: 20px;
            }

            &:hover::before {
                background-color: ${props.theme.data.primaryColor};
            }
        `
    )}
`;