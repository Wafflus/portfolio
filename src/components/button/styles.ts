import { IButtonStyledProps } from "./types";
import styled, { css } from "styled-components";

export const ButtonStyled = styled.a<IButtonStyledProps>`
    align-items: center;
    background-color: ${({ theme }) => theme.data.secondaryColor};
    border-radius: 8px;
    color: ${({ theme }) => theme.data.text.darkColor};
    display: flex;
    font-size: 2rem;
    font-variant: small-caps;
    gap: 16px;
    height: 66px;
    justify-content: center;
    letter-spacing: 0.1rem;
    max-width: 300px;
    min-width: 200px;
    padding: 8px;
    text-decoration: none;
    transition: transform 0.5s;
    width: 100%;

    ${props => props.outline && css`
        background-color: transparent;
        border: 1px solid ${({ theme }) => theme.data.primaryColor};
        color: #93855B;
    `}

    &:hover {
        box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.1);
        transform: scale(102%);
    }
`;