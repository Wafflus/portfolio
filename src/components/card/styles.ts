import styled from "styled-components";

export const CardStyled = styled.article`
    border-radius: 8px;
    padding: 16px;

    &:nth-child(odd) {
        background-color: ${({ theme }) => theme.data.highlightColor};
    }

    &:hover {
        outline: 3px solid ${({ theme }) => theme.data.borderColor};

        & > div > img {
            border: 3px solid ${({ theme }) => theme.data.primaryColor};
            transform: scale(102%);
        }
    }

    ${props => props.theme.breakpoint.above('lg',
        `
            &:nth-child(even) > div {
                flex-direction: row-reverse;
            }
        `
    )}
`;

export const TitleStyled = styled.h3`
    color: ${({ theme }) => theme.data.text.darkColor};
    font-size: 1.5rem;
    font-variant: small-caps;
    font-weight: 600;
    letter-spacing: 0.1rem;
    text-align: center;
    
    ${props => props.theme.breakpoint.above('lg',
        `
            font-size: 2rem;
            text-align: start;
        `
    )}
`;

export const ToolsStyled = styled.h4`
    color: ${({ theme }) => theme.data.text.defaultColor};
    font-variant: small-caps;
    letter-spacing: 0.1rem;
    text-align: center;
`;

export const DescriptionStyled = styled.p`
    color: ${({ theme }) => theme.data.text.defaultColor};
    flex: 1;
    letter-spacing: 0.1rem;
`;