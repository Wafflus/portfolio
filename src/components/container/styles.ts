import { IContainerStyledProps } from "./types";
import styled from "styled-components";

export const ContainerStyled = styled.div<IContainerStyledProps>`
    height: 100%;
    max-width: 1440px;
    padding: ${props => props.verticalPadding} ${props => props.horizontalPadding};
    width: 100%;
    z-index: 200;
`;