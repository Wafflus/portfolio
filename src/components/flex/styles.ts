import { IFlexStyledProps } from "./types";
import styled from "styled-components";

export const FlexStyled = styled.div<IFlexStyledProps>`
    align-items: ${props => props.defaultFlexProps?.alignItems};
    display: flex;
    flex-direction: ${props => props.defaultFlexProps?.flexDirection};
    gap: ${props => props.defaultFlexProps?.gap};
    justify-content: ${props => props.defaultFlexProps?.justifyContent};
    max-width: ${props => props.defaultFlexProps?.maxWidth};
    width: 100%;
    
    ${props => props.theme.breakpoint.above('lg',
        `
            align-items: ${props.largeBreakpointFlexProps?.alignItems};
            flex-direction: ${props.largeBreakpointFlexProps?.flexDirection};
            gap: ${props.largeBreakpointFlexProps?.gap};
            justify-content: ${props.largeBreakpointFlexProps?.justifyContent};
            max-width: ${props.largeBreakpointFlexProps?.maxWidth};
        `
    )}
`;