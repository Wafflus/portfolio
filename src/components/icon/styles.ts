import styled from "styled-components";

import { IImageProps } from "./types";

export const IconStyled = styled.img<IImageProps>`
    width: ${props => props.size}${props => props.unit};
    height: ${props => props.size}${props => props.unit};
    border-radius: 50%;
`;