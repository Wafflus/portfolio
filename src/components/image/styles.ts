import styled from "styled-components";

export const ImageStyled = styled.img`
    border-radius: 8px;
    border: 3px solid transparent;
    height: 100%;
    max-height: 350px;
    max-width: 500px;
    transition: transform 0.5s, border 0.5s;
    width: 100%;
`;