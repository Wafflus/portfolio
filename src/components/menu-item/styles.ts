import { IAnchorProps } from "./types";
import styled, { css } from "styled-components";

export const AnchorStyled = styled.a<IAnchorProps>`
    color: ${props => props.active ? props.theme.data.text.darkColor : props.theme.data.text.defaultColor};
    cursor: pointer;
    display: inline-block;
    font-weight: ${props => props.active ? 'bold' : 'normal'};
    letter-spacing: 0.1rem;
    text-decoration: none;
    transition: transform 0.5s, color 0.5s, padding 0.5s;

    &:hover {
        color: ${({ theme }) => theme.data.text.darkColor};
        text-decoration: none;
    }

    ${props => !props.active && css`
        &:hover {
            transform: translateY(-4px);
        }
    `};

    ${props => props.active && css`
        border-bottom: 1px solid ${({ theme }) => theme.data.text.defaultColor};
        font-size: 1.25rem;
        font-variant: small-caps;
        padding: 4px;

        &::before,
        &::after {
            background-color: ${({ theme }) => theme.data.text.defaultColor};
            border-radius: 50%;
            bottom: -4px;
            content: '';
            height: 8px;
            position: absolute;
            transition: left 0.5s, right 0.5s, background-color 0.5s;
            width: 8px;
        }

        &::before {
            left: -4px;
        }

        &::after {
            right: -4px;
        }

        &:hover {
            border-color: ${({ theme }) => theme.data.text.darkColor};
            
            &::before,
            &::after {
                background-color:${({ theme }) => theme.data.text.darkColor};
            }

            &::before {
                left: -8px;
            }

            &::after {
                right: -8px;
            }
        }
    `}
`;