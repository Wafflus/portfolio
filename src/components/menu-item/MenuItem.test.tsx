import "jest-styled-components";
import { MenuItem } from ".";
import { render } from "@testing-library/react";
import { theme } from "styles";
import { ThemeProvider } from "styled-components";

test('menu item renders', () => {
    const { container } = render(
        <ThemeProvider theme={theme}>
            <MenuItem id="one"><>Menu One</></MenuItem>
        </ThemeProvider>
    );

    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild).toBeInTheDocument();
});

test('should render a list item', () => {
    const menuItem = render(
        <ThemeProvider theme={theme}>
            <MenuItem id="one"><>Menu One</></MenuItem>
        </ThemeProvider>
    );

    const listItem = menuItem.getByRole('listitem');

    expect(listItem).toBeInTheDocument();
})

test('should render an anchor', () => {
    const menuItem = render(
        <ThemeProvider theme={theme}>
            <MenuItem id="one"><>Menu One</></MenuItem>
        </ThemeProvider>
    );

    const anchor = menuItem.getByRole('link');

    expect(anchor).toBeInTheDocument();
});

test('anchor should have different styles when item is active', () => {
    const menuItem = render(
        <ThemeProvider theme={theme}>
            <MenuItem id="one" activeMenuItem="one"><>Menu One</></MenuItem>
        </ThemeProvider>
    );

    const anchor = menuItem.getByRole('link');

    expect(anchor).toHaveStyleRule('color', theme.data.text.darkColor);
    expect(anchor).toHaveStyleRule('font-weight', 'bold');
    expect(anchor).toHaveStyleRule('border-bottom', `1px solid ${theme.data.text.defaultColor}`);
    expect(anchor).toHaveStyleRule('font-size', '1.25rem');
    expect(anchor).toHaveStyleRule('font-variant', 'small-caps');
    expect(anchor).toHaveStyleRule('padding', '4px');

    const beforeOptions = { modifier: '::before' };
    const afterOptions = { modifier: '::after' };
    const hoverOptions = { modifier: ':hover' };
    const hoverAfterOptions = { modifier: ':hover::after' };
    const hoverBeforeOptions = { modifier: ':hover::before' };

    expect(anchor).toHaveStyleRule('background-color', theme.data.text.defaultColor, beforeOptions);
    expect(anchor).toHaveStyleRule('border-radius', '50%', beforeOptions);
    expect(anchor).toHaveStyleRule('bottom', '-4px', beforeOptions);
    expect(anchor).toHaveStyleRule('content', '\'\'', beforeOptions);
    expect(anchor).toHaveStyleRule('height', '8px', beforeOptions);
    expect(anchor).toHaveStyleRule('position', 'absolute', beforeOptions);
    expect(anchor).toHaveStyleRule('transition', 'left 0.5s,right 0.5s,background-color 0.5s', beforeOptions);
    expect(anchor).toHaveStyleRule('width', '8px', beforeOptions);

    expect(anchor).toHaveStyleRule('background-color', theme.data.text.defaultColor, beforeOptions);
    expect(anchor).toHaveStyleRule('border-radius', '50%', afterOptions);
    expect(anchor).toHaveStyleRule('bottom', '-4px', afterOptions);
    expect(anchor).toHaveStyleRule('content', '\'\'', afterOptions);
    expect(anchor).toHaveStyleRule('height', '8px', afterOptions);
    expect(anchor).toHaveStyleRule('position', 'absolute', afterOptions);
    expect(anchor).toHaveStyleRule('transition', 'left 0.5s,right 0.5s,background-color 0.5s', afterOptions);
    expect(anchor).toHaveStyleRule('width', '8px', afterOptions);

    expect(anchor).toHaveStyleRule('left', '-4px', beforeOptions);

    expect(anchor).toHaveStyleRule('right', '-4px', afterOptions);

    expect(anchor).toHaveStyleRule('border-color', theme.data.text.darkColor, hoverOptions);

    expect(anchor).toHaveStyleRule('background-color', theme.data.text.darkColor, hoverBeforeOptions);
    expect(anchor).toHaveStyleRule('left', '-8px', hoverBeforeOptions);
    
    expect(anchor).toHaveStyleRule('background-color', theme.data.text.darkColor, hoverAfterOptions);
    expect(anchor).toHaveStyleRule('right', '-8px', hoverAfterOptions);
});