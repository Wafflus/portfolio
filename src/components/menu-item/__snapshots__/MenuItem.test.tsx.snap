// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`menu item renders 1`] = `
.c1 {
  color: #93855B;
  cursor: pointer;
  display: inline-block;
  font-weight: normal;
  -webkit-letter-spacing: 0.1rem;
  -moz-letter-spacing: 0.1rem;
  -ms-letter-spacing: 0.1rem;
  letter-spacing: 0.1rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: -webkit-transform 0.5s,color 0.5s,padding 0.5s;
  -webkit-transition: transform 0.5s,color 0.5s,padding 0.5s;
  transition: transform 0.5s,color 0.5s,padding 0.5s;
}

.c1:hover {
  color: #756A47;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1:hover {
  -webkit-transform: translateY(-4px);
  -ms-transform: translateY(-4px);
  transform: translateY(-4px);
}

.c0 {
  list-style-type: none;
  position: relative;
}

<li
  class="c0"
>
  <a
    class="c1"
    href="#one"
  >
    Menu One
  </a>
</li>
`;
